Sources/Load.php

```diff
@@ -849,6 +850,94 @@ function loadUserSettings()
 	call_integration_hook('integrate_user_info');
 }
 
+/**
+ * Load minimal user settings from members table.
+ *
+ * @param array $users_ids The users IDs to get the data from.
+ * @return array
+ * @throws Exception
+ */
+function loadMinUserSettings($users_ids = [])
+{
+	global $smcFunc, $modSettings, $language;
+	static $user_settings_min;
+
+	if (empty($user_settings_min))
+		$user_settings_min = array();
+
+	if (empty($users_ids))
+		return array();
+
+	// Already loaded?
+	foreach ($users_ids as $key => $user_id)
+		if (!empty($user_settings_min[$user_id]))
+			unset($users_ids[$key]);
```

Comment:
```suggestion
	$users_ids = array_diff($users_ids, array_keys($user_settings_min));
```

Reply:
i feel overwhelmed by this code of line.